feat(ssh_tunnel): Delete command & exceptions

Antonio-RiveroMartnez opened the pull request:
### SUMMARY
- Adding the new DELETE command and its exception to our SSH Tunnel

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Antonio-RiveroMartnez committed:
SSH Tunnel:

- Add Delete Command and its exceptions

hughhhh commented on superset/databases/commands/delete_ssh_tunnel.py:
```diff
@@ -0,0 +1,52 @@
+        self._model_id = model_id
+        self._model: Optional[SSHTunnel] = None
+
+    def run(self) -> Model:
```
can we write a small unit test for deleting this the same way as the last PR we have

Antonio-RiveroMartnez replied:
Oh yeah! Let me add those for this and the update one

hughhhh commented on superset/databases/commands/exceptions.py:
```diff
@@ -176,3 +176,11 @@ class DatabaseOfflineError(SupersetErrorException):
+
+
+class SSHTunnelDeleteFailedError(DeleteFailedError):
+    message = _("SSH Tunnel could not be deleted.")
```
can you add a status `500` on this one

Antonio-RiveroMartnez replied:
The parent class already has `status = 500` so it should be already there.

Antonio-RiveroMartnez commented on superset/databases/commands/exceptions.py:
```diff
@@ -176,3 +176,11 @@ class DatabaseOfflineError(SupersetErrorException):
+    message = _("SSH Tunnel could not be deleted.")
+
+
+class SSHTunnelNotFoundError(CommandException):
```
Will do 👍

Antonio-RiveroMartnez committed:
SSH Tunnel:

- Move our SSH tunnel DAO, commands, model and exceptions to its own folder inside databases directory
- Adjust some imports with the new paths
- Add tests to our Delete command
